src/common/tags.c

```diff
@@ -162,6 +162,12 @@ gboolean dt_tag_new(const char *name, guint *tagid)
   }
   sqlite3_finalize(stmt);
 
+  if(g_strstr_len(name, -1, "darktable|") == name)
```

Comment:
I fail to understand this part. in dt_tag_new() you clear the memory.dartable_tags table if the tag starts with "darktable|". Right?

Can you add some comments to explain the logic ? No clear to me today and will certainly raises questions later when reviewing/maintaining this part of the code. Thanks.